spec/services/moves/updater_spec.rb

```diff
@@ -48,16 +44,39 @@
       )
     end
 
-    context 'when status updated' do
+    context 'when status changes without an associated allocation' do
       it 'sets `status_updated` to `true`' do
+        updater.call
         expect(updater.status_changed).to be_truthy
       end
     end
 
+    context 'when status changes to cancelled with an associated allocation' do
+      let!(:allocation) { create :allocation, moves_count: 5 }
+      let!(:move) { create :move, :requested, from_location: from_location, allocation: allocation }
+
+      let(:cancellation_reason) { 'other' }
+      let(:status) { 'cancelled' }
+
+      it 'corrects allocation moves_count' do
+        expect { updater.call }.to change { allocation.reload.moves_count }.from(5).to(0)
+      end
+    end
+
+    context 'when status changes to non cancelled with an associated allocation' do
+      let!(:allocation) { create :allocation, moves_count: 5 }
+      let!(:move) { create :move, :cancelled, from_location: from_location, allocation: allocation }
+
+      it 'corrects allocation moves_count' do
+        expect { updater.call }.to change { allocation.reload.moves_count }.from(5).to(1)
+      end
+    end
+
```

Comment:
can you add a test for when there is no allocation associated with the move? unless that's covered in the existing tests

Reply:
The first one is covered by line 47. I'll add another test for a move with an allocation where the status doesn't change.